### PR TITLE
test on windows and mac

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Let's see what happens

Stackoverflow error on Windows :thinking:

### CODE REVIEW

**Comment by michaelosthege** [comment]
The stack overflow 🐛  of `test_sampling.py::TestSamplePriorPredictive::test_density_dist` on Windows is real!

```
(pm3-dev) E:\Source\Repos\pymc3\pymc3\tests>pytest test_sampling.py::TestSamplePriorPredictive::test_density_dist -v
================================================================================================================================================================== test session starts ==================================================================================================================================================================
platform win32 -- Python 3.7.9, pytest-6.1.2, py-1.9.0, pluggy-0.13.1 -- c:\...\envs\pm3-dev\python.exe
cachedir: .pytest_cache
rootdir: E:\Source\Repos\pymc3, configfile: setup.cfg
plugins: cov-2.10.1
collected 1 item

test_sampling.py::TestSamplePriorPredictive::test_density_dist Windows fatal exception: stack overflow

Current thread 0x00004570 (most recent call first):
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 479 in get
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 497 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 884 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 846 in _batch_appends
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 819 in save_list
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 890 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 843 in _batch_appends
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 819 in save_list
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 843 in _batch_appends
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 819 in save_list
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 843 in _batch_appends
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 819 in save_list
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 843 in _batch_appends
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 819 in save_list
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 549 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 885 in _batch_setitems
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 859 in save_dict
  File "c:\...\envs\pm3-dev\lib\site-packages\dill\_dill.py", line 941 in save_module_dict
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 504 in save
  File "c:\...\envs\pm3-dev\lib\pickle.py", line 662 in save_reduce
  ...
```

**Comment by michaelosthege** [comment]
@MarcoGorelli looks like we both made the same mistake in resolving the merge conflict?

I think I have it right now.. Can push it if you want

**Comment by MarcoGorelli** [comment]
It's OK, done :smile: 

Let's see what errors mac / windows throw now...

**Comment by MarcoGorelli** [comment]
seems like mac tests pass!

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: arviz-compatibility
```
Comment (michaelosthege) [inline .github/workflows/windows.yml]: Forgot to change the name?

File: pymc3/util.py [hunk 2]
```diff
@@ -258,6 +259,14 @@ def enhanced(*args, **kwargs):
 # FIXME: this function is poorly named, because it returns a LIST of
 # points, not a dictionary of points.
 def dataset_to_point_dict(ds: xarray.Dataset) -> List[Dict[str, np.ndarray]]:
+    warnings.warn(
+        "dataset_to_point_dict was renamed to dataset_to_point_list and will be removed!.",
```
Comment (michaelosthege) [inline pymc3/util.py]: Two punctuation characters at the end!.

**Review by michaelosthege — CHANGES_REQUESTED** [review]
I just noticed two small glitches, but with them fixed I think we're good to go!

File: .github/workflows/windows.yml (hunk 1 above)
Comment (MarcoGorelli) [reply .github/workflows/windows.yml]: probably, but this one's also testing arviz compatibility:
```
          conda remove arviz -y
          pip install git+git://github.com/arviz-devs/arviz.git
```
maybe we can remove that and just have it as a suite of windows tests (which I presume we'll want to expand once we have these in)

**Review by michaelosthege — APPROVED** [review]

**Comment by twiecki** [comment]
This is amazing!

**Pull request merged**